web3-functions/pyth-oracle-w3f-priceIds/README.md

```diff
@@ -0,0 +1,88 @@
+
+# Gelato Web3 Pyth Functions with Dynamic Configuration
+This directory contains a template for running a web3 function that's an optimized version of [W3F to Pyth](https://github.com/gelatodigital/w3f-pyth-poc-v2/tree/master/web3-functions/pyth-oracle-w3f) where the staleness,price deviation and priceIds are dynamically configurable/loaded from a Github Gist and `updatePriceFeedsIfNecessary` is called instead of `updatePriceFeeds`.
+
+## Funding
+There are two sets of fees that must be paid:
+1. Gelato Fees:
+    The gelato fees are payed with [1Balance](https://docs.gelato.network/developer-services/1balance). 
+    1Balance allows to deposit USDC on polygon and run the transactions on every network.
+
+    To fund 1Balance please visit the [1balance app](https://beta.app.gelato.network/balance) and deposit USDC.
+    (The 1Balance account has to be created with the same wallet/address that will create the Web3 Function task)
+
+    **On testnets, the Gelato fees are fully subsidized by Gelato**
+2. Pyth Fees:
+    Funding works the same as the `W3F to Pyth` example. 
+    The method that updates the price is payable, the update transaction has to include in the msg.value the corresponding fee:
+
+    ```ts
+        /// @notice Update price feeds with given update messages if any of the price feeds need to be updated
+        /// This method requires the caller to pay a fee in wei; the required fee can be computed by calling
+        /// `getUpdateFee` with the length of the `updateData` array.
+        /// Prices will be updated if they are more recent than the current stored prices.
+        /// The call will succeed even if the update is not the most recent.
+        /// @dev Reverts if the transferred fee is not sufficient or the updateData is invalid.
+        /// @param updateData Array of price update data.
+        function updatePriceFeedsIfNecessary(
+            bytes[] calldata updateData,
+            bytes32[] calldata priceIds,
+            uint64[] calldata publishTimes
+        ) external payable;
+    ```
+
+    - W3F Pyth Dynamic Configuration
+       The on-chain transaction executed via a web3 function gets routed through a proxy smart contract which is solely owned by the web3 function task creator (aka the `dedicatedMsgSender`). This `dedicatedMsgSender` proxy contract will be deployed with the first task created by your wallet and will ultimately be responsible for transferring the fee to the Pyth contract.
+**Note that after you create first task, you must ensure the `dedicatedMsgSender` is funded otherwise the task will fail to execute**
```

Comment:
should there be a step 5 below explaining how to fund this address?

Reply:
funding the `dedicatedMsgSender` is just a standard erc 20 transfer but i've updated the README to explicitly state this

Reply:
i think it's good to be explicit -- even if it's easy to do, having a complete checklist will make it easy for people to do the everything needed and not forget steps.